webpack/assets/javascripts/foreman_tools.js

```diff
@@ -10,6 +10,59 @@ import { sprintf, translate as __ } from './react_app/common/I18n';
 
 import { showLoading, hideLoading } from './foreman_navigation';
 
+import store from './react_app/redux';
+import { openConfirmModal as coreOpenConfirmModal } from './react_app/components/ConfirmModal';
+import { noop } from './react_app/common/helpers';
+
+export function openConfirmModal({
+  title = '',
+  message = '',
+  onConfirm = noop,
+  onCancel = noop,
+  modalProps = {},
+  isWarning = false,
+  confirmButtonText = null,
+}) {
+  return store.dispatch(
+    coreOpenConfirmModal({
+      title,
+      message,
+      onConfirm,
+      onCancel,
+      modalProps,
+      isWarning,
+      confirmButtonText,
+    })
+  );
+}
+
+// override the jQuery UJS $.rails.allowAction
+$.rails.allowAction = function railsConfirmOverride(element) {
```

Comment:
do you mean a rails integration test? ok, will try to do something

Reply:
yes. i.e - an integration test that deletes a resource with a visible confirmation modal

Reply:
I tried overriding `confirm` with an async method but looks like it doesn't work since the default rails method returns ` answer && callback`, and when `answer` is an unresolved promise that is considered a truthy value. 

However, looks like the current implementation doesn't follow the API of the rails implementation, which we should maintain to ensure nothing breaks:
https://github.com/rails/jquery-ujs/blob/master/src/rails.js#L287-L296